lab4/src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,59 @@
+package agh.ics.oop;
+
+import java.util.HashMap;
+
+class RectangularMap implements IWorldMap {
+
+    // hashmap which keys are position (as vector) and values are Animal
+    private final HashMap<Vector2d, Animal> animalPositions;
+
+    private final Vector2d UPPER_RIGHT;
+    private final Vector2d LOWER_LEFT = new Vector2d(0, 0);
+    private final MapVisualizer visualizer;
+
+
+    public RectangularMap(int width, int height) {
+        UPPER_RIGHT = new Vector2d(width, height);
+        this.visualizer = new MapVisualizer(this);
+        this.animalPositions = new HashMap<>();
+    }
+
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        return !animalPositions.containsKey(position) && position.precedes(UPPER_RIGHT) && position.follows(LOWER_LEFT);
```

Comment:
ta metoda raczej powinna sprawdzać też kolizje, bo wołamy ją w klasie `Animal` gdy chcemy sie poruszyć na inne pole.

Reply:
Mógłbym Pan uszczegółowić co to znaczy sprawdzać kolizje? W klasie Animal gdy chcemy iść dalej sprawdzamy potencjalne pole właśnie wywołując tą metode która sprawdza !animalPositions.cointainsKey(position) -> gdzie position jest naszą potencjalną pozycją także sprawdza czy pole na ktore chcemy isc nie jest zajęte

Reply:
fakt, nie doczytałem dokładnie, z rozmachu napisałem bo nie ma tu wywołania `isOccupied()`. Właściwie to tak czy inaczej można tego containsa zastąpić wywołaniem `isOccupied()` właśnie żeby nie powtarzać tego samego kodu (i będzie to nieco bardziej czytelne)